sentry/src/main/java/io/sentry/Baggage.java

```diff
@@ -217,10 +218,12 @@ public Baggage(
       final @Nullable Double sampleRate,
       final @Nullable Double sampleRand,
       final @Nullable String thirdPartyHeader,
-      boolean isMutable,
-      boolean shouldFreeze,
+      final boolean isMutable,
+      final boolean shouldFreeze,
       final @NotNull ILogger logger) {
-    this.keyValues = keyValues;
+    // TODO should we deep-copy the keyValues here?
+    // if so we could optimize this by only synchronizing in case isMutable is true
```

Comment:
There are cases where we forcefully overwrite the sample rate, even if mutable = false.